src/Microsoft.Restier.Core/Query/QueryModelReference.cs

```diff
@@ -240,10 +280,19 @@ public IEdmProperty Property
         {
             get
             {
-                var structuredType = this.Source.Type as IEdmStructuredType;
-                if (structuredType != null)
+                if (property != null)
+                {
+                    return property;
+                }
+
+                if (Source != null)
```

Comment:
There are two constructor, if constructor of line 230 is used, then the logic will be called.